### ISSUE TITLE
Backup codes can be downloaded within session

### ISSUE THREAD

**Issue by brynwhyman** [Firesphere/silverstripe-bootstrapmfa#14]
Allow backup codes to be downloaded within the session that they have been created in.

AC:
- [ ] Creates session-unique download link
- [ ] Backup codes are not available outside of the session
- [ ] Allows for the scenario where a user could download backup codes after MFA is set up and codes are generated
- [ ] Allows for the scenario where a user requests a new set of backup codes in their CMS user profile

Notes:
- A separate issue will be raised for how to display a downlink link in the CMS and the login modal

**Comment by brynwhyman** [comment]
Pointing the javascript method.

**Comment by chillu** [comment]
> js method is creating a .txt file

I think this is overkill, just allow printing and copy/paste.

**Comment by robbieaverill** [comment]
I totally agree with @chillu

**Comment by robbieaverill** [comment]
>Allows for the scenario where a user could download backup codes after MFA is set up and codes are generated

I'm not sure this is a "standard" thing to do with MFA flows. Usually you only have one opportunity to get them; when they're first generated and shown to you

**Comment by ScopeyNZ** [comment]
I'll just remove that bit. When we were talking about this last week we were discussing how to avoid having them stored in session ever (for the download link) and everyone seemed on the same page...

**Comment by robbieaverill** [comment]
In terms of the display of the backup codes, it'll happen in two places: the CMS in a Member's CMS fields, and on the frontend once a user has enabled and configured a MFA method (only TOTP, or do backup codes apply to other methods like Yubikey as well?). 

We can/should re-use the logic for generating the CMS fields for both places.

I guess the frontend version would be a new controller action which would be redirected to after successfully configuring MFA. In the CMS it'd be part of CMS fields.

**Comment by ScopeyNZ** [comment]
> do backup codes apply to other methods like Yubikey as well

They do. All MFA methods extending this module will have backup codes provided. But I guess you don't get another set for a second MFA method? 🤔 

Sounds good to me - we just want to avoid ever having backup codes in session.

**Comment by robbieaverill** [comment]
You're right, I'm sorry. Slack lets me view my unused backup codes whenever I want.

**Comment by brynwhyman** [comment]
Replaced with https://github.com/silverstripe/silverstripe-mfa/issues/13
